ui/dashboards/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx

```diff
@@ -32,6 +32,24 @@ import { VariableEditorState, getVariableDefinitionFromState, getInitialState }
 
 const VARIABLE_TYPES = ['ListVariable', 'TextVariable'] as const;
 
+// TODO: Replace with proper validation library
+function getValidation(state: ReturnType<typeof getInitialState>) {
+  /** Name validation */
+  let name = null;
+  if (!state.name) {
+    name = 'Name is required';
+  }
+  // name can only contain alphanumeric characters and underscores and no spaces
+  if (state.name && !/^[a-zA-Z0-9_-]+$/.test(state.name)) {
+    name = 'Name can only contain alphanumeric characters, underscores, and dashes';
```

Comment:
Nice! I’ll probably reuse this tomorrow when doing the form control for optional y_axis label